### ISSUE TITLE
App crashes the first time the user uses the app.

### ISSUE THREAD

**Issue by kastriotrama** [crisp-im/crisp-sdk-android#97]
<img width="758" alt="screen-shot-2022-03-28-at-1748_9p0c3j" src="https://user-images.githubusercontent.com/23016147/160482723-f3ddfd6d-07b6-47af-8f99-70c42e1dbec4.png">

While using the latest crisp sdk(1.0.9), everytime the user tries it for the first time it throws this error, but after it comes back the second time it works, and it won't throw the error even if the user force closes the app.

@Doc1faux @baptistejamin

**Comment by Doc1faux** [comment]
Hi @kastriotrama and thank you for your feedback.

I tried to reproduce your crash creating a brand new Kotlin app (as this is the exception thrown) as is and I didn't reproduced the crash :/
Then, I tried to import up-to-date `socket.io-client`, `okhttp` and/or `retrofit` libraries as it seems to be related to them but I still not reproduce the crash :/ (I say up-to-date as we use previous versions for Socket.IO to be compatible with Crisp servers and OkHttp and Retrofit to be compatible with Android 4.4).
Then I searched specifically for your crash and found [this socket.io-client issue](https://github.com/socketio/socket.io-client-java/issues/611).
At OkHttp, I found [this issue](https://github.com/square/okhttp/issues/5705) too, but the fix does not seem to have been backported to `v3.12.x`...
Could you try to import `v3.14.6` of the OkHttp library in your gradle file to know if it fixes your crash?
```
implementation("com.squareup.okhttp3:okhttp:3.14.6")
```

If so, we should discuss about dropping Android 4.4 @baptistejamin support...

**Comment by Doc1faux** [comment]
Do you deploy your app on a Xiaomi device? Because it seems related to a [firmware issue on MIUI](https://github.com/akexorcist/Localization/issues/89#issuecomment-788181187) while calling `Activity.recreate()` which Crisp SDK does to apply the correct Chat theming according to the website settings.
As I don't own a Xiaomi device to reproduce this crash, I could only provide you a workaround I found which seems to work for some people by disabling auto-dark feature, let me known if some work for you :)
## Disable auto-dark feature
Set `forceDarkAllowed` to `false` to your app theme and to `CrispTheme`
```
<style name="Theme.MyApplication" parent="Theme.MaterialComponents.DayNight.DarkActionBar">
  <item name="android:forceDarkAllowed" tools:targetApi="q">false</item>
</style>

<style name="CrispTheme" parent="Theme.AppCompat.Light.DarkActionBar">
  <item name="android:forceDarkAllowed" tools:targetApi="q">false</item>
</style>
```

**Comment by Doc1faux** [comment]
> updating to the latest `v4.9.2` fixed the issue

So @baptistejamin do we drop the Android 4.4 support to avoid this crash for kotlin developers?

**Comment by baptistejamin** [comment]
@Doc1faux go for it

**Comment by Doc1faux** [comment]
> Tried it with `v3.14.6` but still crashed, updating to the latest `v4.9.2` fixed the issue, thank you for your detailed response @Doc1faux
> 
> Even though it opens up the activity it throws this exception.
> 
> <img alt="Screen Shot 2022-03-29 at 13 36 05" width="1251" src="https://user-images.githubusercontent.com/23016147/160602700-54cc4c96-bb8e-4140-bc79-74e3c955ae04.png">

I moved this crash in new https://github.com/crisp-im/crisp-sdk-android/issues/99 as I close this issue for the fix in `1.0.10` of the first crash due to OkHttp.
